Objects/exceptions.c

```diff
@@ -3852,28 +3848,18 @@ _PyErr_TrySetFromCause(const char *format, ...)
      * types as well, but that's quite a bit trickier due to the extra
      * state potentially stored on OSError instances.
      */
-    /* Ensure the traceback is set correctly on the existing exception */
-    if (tb != NULL) {
-        PyException_SetTraceback(val, tb);
-        Py_DECREF(tb);
-    }
-
     va_start(vargs, format);
     msg_prefix = PyUnicode_FromFormatV(format, vargs);
     va_end(vargs);
     if (msg_prefix == NULL) {
-        Py_DECREF(exc);
```

Comment:
We should still decref `exc` here, right? AFAICS, we still hold a strong reference to it (`PyException_SetCause` swallows it just below).